Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -226,6 +227,64 @@ static std::optional<CSSWideKeyword> consumeCSSWideKeyword(CSSParserTokenRange&
     return keyword;
 }
 
+// MARK: - function value consumer
+
+static bool consumeFunctionArgument(CSSParserTokenRange& range, unsigned index, CSSPropertyID property, CSS::PropertyParserState& state, CSS::PropertyParserResult& result)
+{
+    auto argument = CSSPropertyParserHelpers::consumeArgument(range, index);
+    if (!argument)
+        return false;
+
+    const auto& context = state.context;
+    auto important = state.important;
+    auto ruleType = state.currentRule;
+
+    return consumeStyleProperty(*argument, context, property, important, ruleType, result);
+}
+
+static bool consumeInternalAutoBaseFunction(CSSParserTokenRange& range, CSSPropertyID property, CSS::PropertyParserState& state, CSS::PropertyParserResult& result)
+{
+    // -internal-auto-base() = -internal-auto-base( <value>, <value> )
+
+    ASSERT(range.peek().functionId() == CSSValueInternalAutoBase);
+
+    auto args = CSSPropertyParserHelpers::consumeFunction(range);
+
+    Vector<CSSProperty, 256> autoProperties;
+    CSS::PropertyParserResult autoResult { autoProperties };
+
+    if (!consumeFunctionArgument(args, 0, property, state, autoResult))
+        return false;
+
+    Vector<CSSProperty, 256> baseProperties;
+    CSS::PropertyParserResult baseResult { baseProperties };
+
+    if (!consumeFunctionArgument(args, 1, property, state, baseResult))
+        return false;
+
+    if (autoProperties.size() != baseProperties.size())
+        return false;
+
+    for (unsigned index = 0; index < autoProperties.size(); ++index) {
+        const auto& autoProperty = autoProperties[index];
+        const auto& baseProperty = baseProperties[index];
+
+        Ref value = CSSFunctionValue::create(CSSValueInternalAutoBase, autoProperty.protectedValue(), baseProperty.protectedValue());
+        result.addProperty(CSSProperty(autoProperty.metadata(), WTFMove(value)));
+    }
+
+    return true;
+}
+
+static bool parseInternalAutoBaseFunction(CSSParserTokenRange& range, CSSPropertyID property, CSS::PropertyParserState& state, CSS::PropertyParserResult& result)
+{
+    if (!state.context.cssInternalAutoBaseParsingEnabled)
+        return false;
+    if (range.peek().functionId() != CSSValueInternalAutoBase)
+        return false;
+    return consumeInternalAutoBaseFunction(range, property, state, result);
+}
+
```

Comment:
Do we need both `parseInternalAutoBaseFunction` and `consumeInternalAutoBaseFunction` ? Can they just be a single method?